### PR TITLE
Adding @noreference  ref issue #2481

### PR DESCRIPTION
**Pull request by mpalat**
## Author checklist

- [ ] I have thoroughly tested my changes
- [ ] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [ ] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.core/.settings/.api_filters [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
                 <message_argument value="org.eclipse.jdt.core_3.38.0"/>
             </message_arguments>
         </filter>
     </resource>
-    <resource path="dom/org/eclipse/jdt/core/dom/AST.java" type="org.eclipse.jdt.core.dom.AST">
-        <filter id="388194388">
+    <resource path="dom/org/eclipse/jdt/core/dom/ASTMatcher.java" type="org.eclipse.jdt.core.dom.ASTMatcher">
+        <filter comment="Preview Feature - Class name changed - should not have been an API in the first place " id="338792546">
             <message_arguments>
-                <message_argument value="org.eclipse.jdt.core.dom.AST"/>
```
Comment (iloveeclipse) [inline org.eclipse.jdt.core/.settings/.api_filters:22]: I see API error in IDE: `The constant value 21 of the field org.eclipse.jdt.core.dom.AST.JLS_Latest has been changed`.

**Comment by iloveeclipse** [comment]
Also **I believe** if we are discussing about removal of `AbstractUnnamedTypeDeclaration` that type also should be marked as no APi for now?

**Comment by iloveeclipse** [comment]
@mpalat : build fails in API tooling. 
Could you please fix one API error & 6 warnings reported at https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-2486/1/eclipse/new/ and check my comment regarding `AbstractUnnamedTypeDeclaration`?

**Comment by iloveeclipse** [comment]
> Looks like a diff in my config locally with Jenkins - am at baseline 4.31 locally and I had to add those filters - investigating why there is a difference between the two

If you are still using Eclipse 4.31, you have to update to a latest 4.32 build, as PDE API tooling needs fixes (included in 4.32) for broken 4.31 api descriptions shipped with 4.31

**Comment by mpalat** [comment]
> https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-2486/1/eclipse/new/

Thanks @iloveeclipse for the suggestion. I was working with 4.32 albeit an older version - so downloaded yesterday's version - unfortunately I don't see a difference - meanwhile updated the PR with the fix for some of the warnings - not the API ones - investigating that

**Comment by iloveeclipse** [comment]
@mpalat : I wonder if you could review https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2483, which seem to be more clean & complete solution ?

**Comment by mpalat** [comment]
> #2483

Hmm.. Good that the original author himself is attempting the fix - I will abandon this PR then [@srikanth-sankaran  - please note.]

**Comment by mpalat** [comment]
Abandoning this PR for https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2483, - please see comment above

**Pull request closed without merging**